src/net.cpp

```diff
@@ -997,17 +997,29 @@ std::vector<uint8_t> GenerateRandomGarbage() noexcept
 
 } // namespace
 
-V2Transport::V2Transport(NodeId nodeid, bool initiating, int type_in, int version_in, const CKey& key, Span<const std::byte> ent32, Span<const uint8_t> garbage) noexcept :
+void V2Transport::StartSendingHandshake() noexcept
+{
+    AssertLockHeld(m_send_mutex);
+    Assume(m_send_state == SendState::AWAITING_KEY);
+    // Initialize the send buffer with ellswift pubkey + provided garbage.
```

Comment:
73ad8bbcd23c6bdf3bde0481ae39633ba5b38c1e: maybe add an Assume that the send buffer was still empty at this point